syntax="proto3";

import "common.proto";

package login;

message LoginMsg {                              // 奇数是请求，偶数是响应
    oneof choice {
        common.TestReq  test_req        = 1;    // 请求
        common.TestRsp  test_rsp        = 2;    // 响应

        LoginReq        login_req       = 3;    // 登录
        LoginRsp        login_rsp       = 4;

        AccessKeyReq    access_key_req  = 5;    // 获取access_key
        AccessKeyRsp    access_key_rsp  = 6;
    }
}

enum ErrorCode {
    SUCCESS                         = 0;

    // 系统错误码 (各个服务service_id * 10000 + 公共系统错误码)
    ERR_SYS_BEGIN                   = 105000100;
    ERR_SYS_OVERLOAD                = 105000101;    // 服务过载
    ERR_SYS_REJECT_SERVICE          = 105000102;    // 拒绝服务
    ERR_SYS_SERVER_INNER            = 105000103;    // 服务内部错误
    ERR_SYS_TIMEOUT                 = 105000104;    // 超时错误
    ERR_SYS_NO_INSERVICE_LIST       = 105000105;    // 没有可用服务ip
    ERR_SYS_TASK_STATE              = 105000106;    // 任务状态错误
    ERR_SYS_TASK_DISCARD            = 105000107;    // 任务丢弃
    ERR_SYS_END                     = 105000199;

    // 数据包错误码 (各个服务service_id * 10000 + 公共数据包错误码)
    ERR_PACKET_BEGIN                = 105000200;
    ERR_PACKET_ENCODE               = 105000201;   	// 打包失败
    ERR_PACKET_DECODE               = 105000202;   	// 解码失败
    ERR_PACKET_HEADER               = 105000203;
    ERR_PACKET_LEN                  = 105000204;
    ERR_PACKET_VERSION              = 105000205;
    ERR_PACKET_FROM_SERVICE_ID      = 105000206;
    ERR_PACKET_TO_SERVICE_ID        = 105000207;
    ERR_PACKET_APP_ID               = 105000208;
    ERR_PACKET_APP_VERSION          = 105000209;
    ERR_PACKET_CONN_SEQ_ID          = 105000210;
    ERR_PACKET_MSG_SEQ_ID           = 105000211;
    ERR_PACKET_DATA_FORMAT          = 105000212;
    ERR_PACKET_DATA_FIELD_0         = 105000213;
    ERR_PACKET_DATA_FIELD_1         = 105000214;
    ERR_PACKET_DATA_FIELD_2         = 105000215;
    ERR_PACKET_DATA_FIELD_3         = 105000216;
    ERR_PACKET_CHECK_SUM            = 105000217;
    ERR_PACKET_UNKNOWN_REQUEST      = 105000218;   	// 不明请求
    ERR_PACKET_END                  = 105000299;

    // 接口错误码 (各个服务service_id * 10000 + 公共接口错误码)
    ERR_INTERFACE_BEGIN             = 105000300;
    ERR_INTERFACE_PARAM             = 105000301;   	// 参数错误
    ERR_INTERFACE_PERM              = 105000302;   	// 权限错误
    ERR_INTERFACE_TIMEOUT           = 105000303;   	// 接口超时
    ERR_INTERFACE_END               = 105000399;

    // 业务错误码 (各个服务service_id * 10000 + 公共业务错误码)
    ERR_BUSINESS_BEGIN              = 105001000;
    ERR_BUSINESS_END                = 105009999;
}

message LoginReq {
    int32   service_id          = 1;    // navigate::GetAccessRsp.service_id
    uint32  proc_id             = 2;    // navigate::GetAccessRsp.proc_id
    uint64  user_id             = 3;
    bytes   passwd              = 4;    // 字符串不安全，抓包显示出来太明显
    uint32  dev_type            = 5;
    bytes   dev_token           = 6;
}

message LoginRsp {

}

message AccessKeyReq {

}

message AccessKeyRsp {
    bytes   access_key          = 1;
}

